### ISSUE TITLE
work on eliminating some BC Breaks

### ISSUE THREAD

**Issue by craigh** [zikula/core#1856]
refs #1853 

There seems to be some possibility to remove BC breaks in the areas that are currently in the documentation. I suggest we work on seeking and implementing BC where possible.

 - [x]  $this->request->files
 - [x]  $this->request->filter
 - [x]  Paginate (Doctrine Extensions)
 - [x]  Events

$this->request->files
----
note: there is still some question as to whether the old ->getFiles() method might be able to maintain BC or to find another method of providing for BC in this case. (refs #1002, #1261) "one could replace the ParameterBag with a proxy class and magic methods for example. The request object is supposed to be created only once in the front controller btw"

$this->request->filter
----
note: There may still be some effort to provide a BC layer here. (refs #1261) "if someone cares enough it might be possible to handle with a proxy"

Paginate (Doctrine Extensions)
----
note: is this a true BC-break?

Events
----
Could the old events be triggered by the new events for BC purposes only?

**Comment by craigh** [comment]
events were added back in #2050

**Comment by craigh** [comment]
@cmfcmf @Guite can either of you explain what the issue is with **Paginate (Doctrine Extensions)** ? What happened, what was required to be changed and why?

**Comment by Guite** [comment]
Doctrine 2.2 introduced a Paginator component:
http://doctrine-orm.readthedocs.org/en/latest/tutorials/pagination.html

In the 1.3 core we included Doctrine 2.1.4
https://github.com/zikula/core/blob/1.3/src/plugins/Doctrine/Plugin.php#L59
so we had to use an extension for this:
https://github.com/beberlei/DoctrineExtensions#paginator

More information with regards to the different implementations can be found here:
https://github.com/Guite/MostGenerator/blob/master/org.zikula.modulestudio.generator/src/org/zikula/modulestudio/generator/cartridges/zclassic/models/Repository.xtend#L203
https://github.com/Guite/MostGenerator/blob/master/org.zikula.modulestudio.generator/src/org/zikula/modulestudio/generator/cartridges/zclassic/models/Repository.xtend#L732

**Comment by craigh** [comment]
@Guite what about including the Extension in 1.4.0 so both options are available?

**Comment by Guite** [comment]
I wouldn't do this.
First the repo isn't maintained:
```
Warning: This repository is not really maintained anymore. The important paginatior and large collections code was moved to the Doctrine2 core. For all the other extensions, be aware that I don't maintain them anymore.
```
Second how many 1.3 modules are using it actually?
MOST-based modules can be regenerated with ease anyway, so they don't really count.

**Comment by Guite** [comment]
Also we don't know whether the extension works with Doctrine 2.2+. So if we would really want to provide a BC layer, we would probably need to include the old Doctrine version, too - which is going to become ugly if we need to have two Doctrine installations in parallel.

**Comment by craigh** [comment]
I was just trying to find ways to keep BC as much as possible. Even though it isn't maintained, we might include it but only if we can confirm it is compatible with D2.2+ (if it is not used, all the better :smile: ).

I agree we should not include two versions of doctrine.

**Comment by craigh** [comment]
:laughing: :laughing: :laughing: 

The extension is ***still included*** already :exclamation:

**Comment by cmfcmf** [comment]
I think I re-included it after discussion with @drak. You should find a PR.

**Comment by craigh** [comment]
IMO, including it is enough to clear the BC issue.

**Comment by craigh** [comment]
original ticket is #890. PR is #891.

**Comment by craigh** [comment]
I've updated the docs for a more accurate representation of that concern in my pending PR.

If anyone is aware of any other vendor library that is also deprecated, then we should also add it to the docs.

IMO when my PR is merged, this ticket is complete.

**Comment by craigh** [comment]
closed by #2214
